### PR TITLE
Source template variable restrictions

### PR DESCRIPTION
**Pull request by leijou**
Related to #109 - allows a template file to act as the base list of included environment variables, and optionally enforce them all as strict requirements.

See README.md changes for documentation.

I've not used poetry before and was unable to get it running in my environment, so I was unable to run all of the steps in the contributing guide. Please forgive me :pray: I've done my best to match surrounding styles

### CODE REVIEW

**Comment by leijou** [comment]
Build failures are all now complexity based. Could we up the thresholds?
```
./dump_env/dumper.py
  0:1      WPS226 Found string constant over-use: '' > 3
  89:1     C901  'dump' is too complex (7)
```

File: README.md [hunk 1]
```diff
@@ -82,6 +82,45 @@ VAR=abc
 Any number of `--strict` flags can be provided.
 No more forgotten template overrides or missing env vars!
 
+### Master templates
```
Comment (sobolevn) [inline README.md:85]: Let's call it `source`
It would be more clear. And `master` is a hot topic right now.

**Comment by sobolevn** [comment]
@leijou I have fixed the CI, can you please rebase your code?

**Comment by leijou** [comment]
Rebased & squashed. Still failing on complexity thresholds

**Comment by sobolevn** [comment]
Can you please then refactor the places it reports as complex ones? 
We have a contribution guide where you can find how to use `poetry` and all other tooling we have: https://github.com/sobolevn/dump-env/blob/master/CONTRIBUTING.md

**Comment by sobolevn** [comment]
👀 

Thanks a lot! Will review it today! 👍

**Pull request merged**